webrender/src/frame_builder.rs

```diff
@@ -882,12 +882,17 @@ impl FrameBuilder {
                     BoxShadowClipMode::Inset => 1.0,
                 };
 
-                // If we have a border radius, we'll need to apply
-                // a clip-out mask.
+                // Outset box shadows with border radius
+                // need a clip out of the center box.
+                let extra_clip_mode = match clip_mode {
```

Comment:
What I mean is, instead of selecting a different clip mode, I think it would work (and be more GPU efficient) if you change the ```if border_radius > 0.0``` condition to ```if border_radius > 0.0 && clip_mode == BoxShadowClipMode::Inset``` ?